src/tattoo.ts

```diff
@@ -333,29 +386,55 @@ async function analyzeRepos() {
   return analyzer;
 }
 
-function openRepo(cloneOptions: nodegit.CloneOptions, ghRepo: GitHub.Repo) {
+
+
+async function openRepo(cloneOptions: nodegit.CloneOptions,
+  ghRepo: GitHub.Repo,
+  branchConfig: BranchConfig): Promise<ElementRepo> {
   const dir = path.join("repos", ghRepo.name);
-  let repoPromise: Promise<nodegit.Repository>;
+  let repo: nodegit.Repository;
   if (util.existsSync(dir)) {
     let updatedRepo: nodegit.Repository;
-    repoPromise = nodegit.Repository.open(dir).then((repo) => {
+    repo = await nodegit.Repository.open(dir).then((repo) => {
         updatedRepo = repo;
         return cloneRateLimiter.schedule(() =>
           updatedRepo.fetchAll(cloneOptions.fetchOpts)
         );
       }
     ).then(() => updatedRepo);
   } else {
-    repoPromise = cloneRateLimiter.schedule(() => {
+    repo = await cloneRateLimiter.schedule(() => {
```

Comment:
there's a potential race condition here that you probable will never hit, but it's possible for the repo directory to not exist, but for the to be a clone operation in flight. Depending on if `openRepo` is ever called with the same repo, this might not be a worry. Otherwise a map of clone operations would protect against it.